blamo/lib/Test/test.dart

```diff
@@ -51,29 +51,96 @@ class _TestPageState extends State<TestPage> {
                   FormBuilderTextField(
                     attribute: 'type-no',
                     validators: [],
-                    decoration: InputDecoration(labelText: "Type Number"),
+                    decoration: InputDecoration(labelText: "Test Type And Number"),
+                  ),
+                  FormBuilderTextField(
+                    attribute: 'tst-depth-ub',
+                    validators: [FormBuilderValidators.numeric()],
+                    decoration: InputDecoration(labelText: "Depth Upper Bound (m)"),
+                  ),
+                  FormBuilderTextField(
+                      attribute: 'tst-depth-lb',
+                      validators: [FormBuilderValidators.numeric()],
+                      decoration: InputDecoration(labelText: "Depth Lower Bound (m)")
                   ),
                   FormBuilderTextField(
                     attribute: 'recovery-%',
                     validators: [FormBuilderValidators.numeric()],
-                    decoration: InputDecoration(labelText: "Recovery %"),
+                    decoration: InputDecoration(labelText: "Percent Recovery"),
                   ),
                   FormBuilderTextField(
                     attribute: 'sdr-rdd-rqd',
                     validators: [],
                     decoration: InputDecoration(labelText: "Soil Driving Resistance / Rock Discontinuity Data or RQD"), //ASK - preferred title?
                   ),
+                  FormBuilderTextField(
+                    attribute: 'prcnt-moisture',
+                    validators: [],
+                    decoration: InputDecoration(labelText: "Percent Natural Moisture"),
+                  ),
+                  FormBuilderTextField(
+                    attribute: 'user-material-desc',
+                    validators: [],
+                    decoration: InputDecoration(labelText: "Material Description"),
+                  ),
                   FormBuilderCheckboxList( //TODO - redirect to longer comprehensive list of tags? Refactor to a list of autocompleting text fields? (SEE: unit.dart, 51)
                     attribute: 'material-description',
                     validators: [],
                     initialValue: [],
                     options: [ // TODO need gint's set of tags, ability for user to make own tags.
-                      FormBuilderFieldOption(value: "Sandy GRAVEL (Shoulder Aggregate)"),
-                      FormBuilderFieldOption(value: "GP"),
-                      FormBuilderFieldOption(value: "Varlegated Gray"),
-                      FormBuilderFieldOption(value: "Nonplastic"),
-                      FormBuilderFieldOption(value: "Moist"),
-                      FormBuilderFieldOption(value: "(FILL)")
+                      FormBuilderFieldOption(value: "Asphalt"),
```

Comment:
From what i see most everything seems functional. We might want to remove the .idea folder, and maybe consider looking into automating some of the form builder options. It looks like its a list of FormBuilderFieldOption so maybe making all the options into a list of strings, and iteratively adding them to the List of  FormBuilderFieldOption.

Reply:
I was looking at potentially using this [this field type](https://pub.dev/packages/flutter_typeahead) so that we can take user input as well as offer the list of predefined soil types. And since the soil types we were given all have associated abbreviations i think this format could work well.

Reply:
The field type option you linked would be really nice. I think then. for now just remove the .idea and fix the index and you'll have my goahead.